### ISSUE TITLE
private vs. protected in visitor / deparser

### ISSUE THREAD

**Issue by ted-johnson** [JSQLParser/JSqlParser#675]
Is there a reason why deparsers and the buffer are private in the visitor and deparser classes?
Shouldn't they be protected instead?

For example, I would like to insert some extra output during Insert statement deparsing.
But the StatementDeParser class defines its members as:
     public class StatementDeParser implements StatementVisitor {
         private ExpressionDeParser expressionDeParser;
         private SelectDeParser selectDeParser;
         private StringBuilder buffer;
The InsertDeParser defines these members similarly.

If I try to write
    @Override
    public void visit(Insert insert) {
        buffer.append("-- an insert statement");
        ...
    }
I get an error saying that buffer is private.
There is a getBuffer method, but using it makes the code cumbersome.
There is no get method for expressionDeParser or selectDeParser, so its much harder
to write code which accesses or manipulates these members.

The value of the visitor and deparser framework would seem to be the ability to modify
selected parts of the deparsing by subclassing and overridding selected methods.
Since these classes are supposed to be subclassed, shouldn't important member variables
be protected instead of private?

**Comment by wumpz** [comment]
The deparsers are unfortunately the way to go here. A better way to manipulate the syntax tree is to use the visitor adapter classes. But they contain far less visit logic (at the moment).

**Comment by ted-johnson** [comment]
The example I wrote has the effect of inserting a comment, and then would access the deparser.
But the deparser is written similarly:
  public class InsertDeParser implements ItemsListVisitor {
      private StringBuilder buffer;
      private ExpressionVisitor expressionVisitor;
      private SelectVisitor selectVisitor;

I'm asking if these member fields are intentionally private,
or if perhaps they can be made protected?

(I did see the src/main/java/net/sf/jsqlparser/util/SelectUtils.java and other examples).
